### PR TITLE
ft(#4): Reset password

### PR DESCRIPTION
**Pull request by munezeromicha**
### What does this PR do?

This pull request implements a new feature that allows users to reset their passwords by sending a password reset email. When a user clicks the reset password link in the email, the app will open and redirect them to the setNewPassword screen.

### Description of Task to be completed?

- Create a new password reset endpoint: Implement a new endpoint that handles password reset requests. This endpoint should generate a unique password reset token and send an email to the user containing a link to a password reset page.
- Implement password reset functionality: Create a new screen or page where users can enter their new password. When a user submits the new password, the app should verify the password reset token and update the user's password if the token is valid.
- Handle email delivery: Integrate an email service (e.g., SendGrid, Mailgun) to send password reset emails. Ensure that the email template includes a clear link to the password reset page.
- Implement security measures: Implement security measures to protect password reset tokens and prevent unauthorized password resets. Consider using techniques like token expiration, token invalidation, and rate limiting.

### How should this be manually tested?

1. Create a new user account.
2. Request a password reset.
3. Click the password reset link in the email.
4. Enter a new password on the setNewPassword screen.
5. Verify that the password reset is successful by logging in with the new password.
6. Test the password reset functionality with invalid email addresses, expired tokens, and rate limiting.

### Screenshots:
<div style="display: flex; flex-wrap: wrap;">
    Screenshot 1
        Screenshot 1
    Screenshot 2
        Screenshot 1
</div>

### CODE REVIEW

**Comment by fredshema** [comment]
Please change your PR title to match the template I shared

**Comment by munezeromicha** [comment]
> Please change your PR title to match the template I shared

**Done**

File: app/(onboarding)/index.tsx [hunk 1]
```diff
@@ -78,10 +78,8 @@ export default function AppOnboarding() {
         <View className={`rounded-full w-4 h-4 ${getDotColor(2)}`}></View>
       </View>
       <View className={`flex-1 flex-row justify-center items-center ${bgColor}`}>
-        <TouchableOpacity>
-          <Text className="text-lg font-Inter-Medium" onPress={() => router.push('/auth/login')}>
-            Get Started
-          </Text>
+        <TouchableOpacity onPress={() => router.push('/auth/forget/reset-password')}>
```
Comment (fredshema) [inline app/(onboarding)/index.tsx]: Use forgot password on login page to link to password reset screens

File: validations/login.schema.ts [hunk 2]
```diff
@@ -4,7 +4,15 @@ export const UserLoginSchema = Yup.object().shape({
   email: Yup.string().email('Invalid email').required('Required'),
   password: Yup.string().min(8, 'Password must be at least 8 characters').required('Required'),
 });
-
+export const ResetPasswordSchema = Yup.object().shape({
+  email: Yup.string().email('Invalid email').required('Required'),
```
Comment (fredshema) [inline validations/login.schema.ts]: Use coherent error message for required rule. eg: `Please provide an email address`

Comment (fredshema) [reply validations/login.schema.ts]: Do the same for the rest

**Review by fredshema — CHANGES_REQUESTED** [review]

**Pull request closed without merging**